src/app/address/validators/addressValidator.js

```diff
@@ -0,0 +1,42 @@
+module.exports = {
+  flatNumberLength: function (val) {
+    return val?.length <= 30;
+  },
+  houseNumberLength: function (val) {
+    return val?.length <= 10;
+  },
+  houseNameLength: function (val) {
+    return val?.length <= 50;
+  },
+  streetNameLength: function (val) {
+    return val?.length <= 60;
+  },
+  localityLength: function (val) {
+    return val?.length <= 60;
+  },
+  alphaNumericWithSpecialChars: function (val) {
+    if (!val) {
+      return true; // handle when field is optional
+    }
+    return !!val.match(/^[0-9A-Z'\\/* -]+$/i);
+  },
+  isValidDate: function (val) {
+    // match 4 digits.
+    // validates, required + is digits + 4 chars long
+    return val.match(/^\d{4}$/) ? true : false;
+  },
+  isPreviousDate: function (val) {
+    const date = new Date(val);
+    if (date.getFullYear() > new Date().getFullYear()) {
+      return false;
+    } else {
+      return true;
+    }
+  },
+  validateHouseNumberAndName: function (houseName, houseNumber) {
+    if (!houseName.trim() && !houseNumber.trim()) {
```

Comment:
same as above, unless am missing something pecular stuff that means you need to do it this way